backend/src/repository/ai_models.py

```diff
@@ -47,7 +48,7 @@ def encode_string(self, data):
             model=self.encoder_model,
             tokenizer=self.encoder_tokenizer,
             prompt=data,
-            device="cpu",
+            device=Devices.get_device().value,
```

Comment:
We check the host devices, it can be mps, cuda and cpu. Since, we split GPU and CPU deployment by using docker-compose. If docker-compose with GPU can be deployment, we will use GPU.  Otherwise, if we use CPU tag docker compose, it should be use CPU as computing device.

Although we support mps, but I do not need to pay too much attention on it. Let's focus on CPU and GPU first.

Reply:
We have two different docker-compose files for CPU and GPU,   the make file was added with make gpu to used the docker-compose.gpu.yaml.

The device.get_device().value here is to adapt the the yaml file with one set of code